src/main/java/randoop/generation/CoverageTracker.java

```diff
@@ -0,0 +1,200 @@
+
+  /**
+   * Coverage details related to a single method under test. Tracks total number of branches and
+   * number of uncovered branches.
+   */
+  public static class CoverageDetails {
+    /** Total number of branches. */
+    public int numBranches;
+    /** Number of uncovered branches. */
+    public int uncoveredBranches;
+  }
+
+  private CoverageTracker() {
+    this.memoryClassLoader = new MemoryClassLoader(this);
+    this.instrumenter = new Instrumenter(runtime);
+    this.data = new RuntimeData();
+
+    try {
+      this.runtime.startup(data);
+    } catch (Exception e) {
+      e.printStackTrace();
+      System.exit(1);
+    }
+  }
+
+  /**
+   * Copy the set of names of classes under test.
+   *
+   * @param classesUnderTest names of all the classes under test.
+   */
+  public void setClassesUnderTest(Set<String> classesUnderTest) {
+    this.classesUnderTest = new HashSet<>(classesUnderTest);
+  }
+
+  /**
+   * Attempts to instrument a class and return the byte array representation.
+   *
+   * @param className name of the class to instrument.
+   * @return the instrumented bytes of the class, null if instrumentation failed.
+   */
+  public byte[] instrumentClass(String className) {
+    // Only instrument classes under test and their nested classes.
+    if (!classesUnderTest.contains(className)) {
+      boolean isNestedClass = false;
+
+      // Determine if this class is a nested class, in which case, the name of some
+      // existing class under test would be the prefix of this class's name.
+      for (String classUnderTest : classesUnderTest) {
+        if (className.startsWith(classUnderTest)) {
+          isNestedClass = true;
+          break;
+        }
+      }
+
+      // If it's not explicitly a class under test, or a nested class of one, don't instrument the class.
+      // Examples include java.lang classes.
+      if (!isNestedClass) {
+        return null;
+      }
+    }
+
+    final byte[] instrumented;
+    final String resource = '/' + className.replace('.', '/') + ".class";
+    InputStream original = getClass().getResourceAsStream(resource);
+
+    if (original == null) {
+      System.err.println("No resource with name: " + resource + " found!");
+      return null;
+    }
+
+    try {
+      // Instrument the class to prepare for coverage collection later.
+      instrumented = instrumenter.instrument(original, className);
+      original.close();
+    } catch (IOException e) {
+      e.printStackTrace();
+      return null;
+    }
+
+    return instrumented;
+  }
+
+  /**
+   * Returns the instrumented version of the class.
+   *
+   * @param className name of the class
+   * @return {@code Class} object that has been instrumented for coverage data collection. Returns
+   *     null if class with target name cannot be found.
+   */
+  public Class<?> getInstrumentedClass(String className) {
+    if (className == null || className.isEmpty()) {
```

Comment:
Should clients pass `""`?  It's not documented as a possibility.  I would drop the disjunct that tests for it.